src/dune/lib.ml

```diff
@@ -682,13 +682,12 @@ module Vlib : sig
   end
 end = struct
   module Unimplemented = struct
-    type status =
-      | Implemented
-      | Not_implemented
-
-    type t = status Map.t
+    type t =
+      { implemented : Set.t
+      ; unimplemented : Set.t
+      }
```

Comment:
No strong objection on this change, but the old representation had the advantage that by construction it was impossible to have a library in both state. Now it is a property that has to be enforced by the code. IMO, the old representation was better.

Reply:
The advantage of this representation is that we loop only over the unimplemented libraries in `handle`. To me that seemed important enough optimize as it basically means that our implementation is quadratic in the number of virtual libraries.

Reply:
Actually, if you think this extra speed doesn't matter i'll just revert. I'm beginning to wonder if it's really necessary.

Reply:
Ok, yh that makes sense. Especially now that we iterate twice over the list

Reply:
The truth is, in practice I don't think it matters too much unless one uses a lot of virtual libraries.

Reply:
So yh I'd just keep the simplest possible implementation

Reply:
Okay, I'll just revert in master.